persistent-test/src/DerivePersistTest.hs

```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE UndecidableInstances, DeriveGeneric #-} -- FIXME
+
+module DerivePersistTest where
+
+import Database.Persist.TH (mkDeleteCascade, derivePersist, persistLowerCase, stripEntityNamePrefix, DeriveEntityDef(..), DeriveFieldDef(..), DeriveForeignKey(..), DeriveUniqueDef(..), mkDeriveEntityDef, mkDeriveFieldDef, mkDeriveUniqueDef)
+import Database.Persist.Quasi (lowerCaseSettings)
+import Data.Maybe (isJust)
+
+import Init
+
+data DerivePrimary = DerivePrimary {
```

Comment:
This module has copies of several tests, covering composite keys, uniques, and the overrides for instance definitions. It would be great to run make whole test suite work with both quasi-quotation and derivePersist, but it's not clear how to automate this.

Reply:
Creation of `EntityDef` is difficult to extract from the rest of `mkPersistWith`. Comparing `UnboundEntityDef`s created by `parse` and `datatypeToEntityDef` would work though. That would require two definitions, but that's more robust than the current DerivePersistTest.